pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -499,18 +499,17 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation) *gardencorev1
 }
 
 func removeTaskAnnotation(ctx context.Context, o *operation.Operation, generation int64, tasksToRemove ...string) error {
-	newShoot, err := kutil.TryUpdateShootAnnotations(ctx, o.K8sGardenClient.GardenCore(), retry.DefaultRetry, o.Shoot.Info.ObjectMeta,
-		func(shoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.Shoot, error) {
-			if shoot.Generation == generation {
-				controllerutils.RemoveTasks(shoot.Annotations, tasksToRemove...)
-			}
-			return shoot, nil
-		},
-	)
-	if err != nil {
+	// Check if shoot generation was changed mid-air
```

Comment:
```suggestion
	// Check if shoot generation was changed mid-air
	// if that is the case, we do not remove the task annotation so that the action
	//  associated with the task is executed again in the subsequent reconciliation
```